source/agora/consensus/Ledger.d

```diff
@@ -228,7 +228,8 @@ public class NodeLedger : Ledger
         if (auto err = this.fee_man.getTxFeeRate(tx, this.utxo_set.getUTXOFinder(),
             &this.getPenaltyDeposit, fee_rate))
             return err;
-        if (fee_rate < min_fee)
+        const size_budget = this.params.MaxTxSetSize * 1024;
+        if (fee_rate < min_fee && this.pool.getPoolSize() >= size_budget) // If pool is sparsely populated, accept any fee.
```

Comment:
Isn't there a risk that no fees will be paid with this change?

Reply:
I will add a test for it.

Reply:
added a simple check for it, FeeManager handles that and does not allow any TXs with fees below `ConsensusConfig.min_fee`